### PR TITLE
Updates to the MainMenu and UserValue UI States

### PR DESCRIPTION
**Pull request by TaylorSmith28**

### CODE REVIEW

File: titration/ui_state/user_value/buffer_ph.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def save_value(self, value):
         """
         The function to save the titrator's buffer pH
         """
-        self.titrator.buffer_ph = value
+        self.titrator.buffer_ph = float(value)
```
Comment (prestoncarman) [inline titration/ui_state/user_value/buffer_ph.py:16]: Should this be done in the UserValue class? Or do you want to ensure its a float by having it here?

Comment (TaylorSmith28) [reply titration/ui_state/user_value/buffer_ph.py:16]: I think this should be done in the UserValue class. I'll fix this shortly.

Comment (TaylorSmith28) [reply titration/ui_state/user_value/buffer_ph.py]: This is fixed, ready to be merged

File: titration/ui_state/user_value/user_value.py [hunk 2]
```diff
@@ -52,9 +52,10 @@ def handle_key(self, key):
         Parameters:
             key (char): the keypad input to determine which state to go to
         """
-        if key == "A":
+        if (key == "A") & (self.value not in ("", ".")):
```
Comment (prestoncarman) [inline titration/ui_state/user_value/user_value.py:55]: What does the "&" do here? I thought python used "and".

File: test/ui_state/user_value/buffer_ph_test.py [hunk 3]
```diff
@@ -167,7 +171,34 @@ def test_buffer_ph(print_mock, set_next_state_mock):
...
+    buffer_ph.handle_key("A")
+    set_next_state_mock.assert_not_called()
+    assert buffer_ph.value == ""
+
+    buffer_ph.loop()
+    print_mock.assert_has_calls(
+        [
+            mock.call("Enter buffer pH:", line=1),
+            mock.call("", style="center", line=2),
+            mock.call("* = .       B = BS", line=3),
```
Comment (prestoncarman) [inline test/ui_state/user_value/buffer_ph_test.py:183]: Can we instructions? What do you think of this message:

```text
 *=. A)ccept  B)ack
     D)ecline C)lear
```

File: titration/ui_state/user_value/user_value.py (hunk 2 above)
Comment (TaylorSmith28) [reply titration/ui_state/user_value/user_value.py]: Yes, I fixed this and implemented it with "and" instead.

File: test/ui_state/user_value/buffer_ph_test.py (hunk 3 above)
Comment (TaylorSmith28) [reply test/ui_state/user_value/buffer_ph_test.py]: I like it. It is implemented

**Pull request merged**